src/Shared/RequestDataHelper.cs

```diff
@@ -26,6 +27,7 @@ internal sealed class RequestDataHelper
     private const string OtherHttpMethod = "_OTHER";
 
     private static readonly char[] SplitChars = new[] { ',' };
+    private static readonly ConcurrentDictionary<string, string> DisplayNameCache = new ConcurrentDictionary<string, string>();
```

Comment:
Should be the size limit for this collection?

Reply:
I don't expect this collection to grow to more than a few hundred items (which is already a lot).

Reply:
You're right, the dictionary will grow without limit. Let me try to thing about a solution.

Reply:
My understanding is that there's no built-in "bounded" collection in .NET. While it's possible to implement a custom one, probably by wrapping `ConcurrentDictionary` with additional checks, I'm not entirely enthusiastic about it. I'm open to suggestions.

Reply:
@joegoldman2 - One possible option is to check the Count of items within the dictionary before adding a new item. The Count could be some pre-defined number. If the Count exceeds that number, then we could simply skip adding items to the Dictionary.